tasks/main.yml

```diff
@@ -97,6 +91,41 @@
     enabled: True
     state: started
 
+- name: Delete existing unwanted network interfaces
+  shell: |
+    if [ ! -e /sys/class/net/{{ interface_name | quote }} ]; then
+      exit 128
+    fi
+    ip link delete {{ interface_name | quote }}
+  # From `man ip`:
+  # Exit status is 0 if command was successful, and 1 if there is a syntax error.  If an error was reported by the kernel exit status is 2.
+  register: result
+  failed_when: result.rc not in [0, 128]
+  changed_when: result.rc != 128
+  notify:
+    - Restart systemd-networkd service
+  loop: "{{ systemd_network_netdevs_to_delete }}"
+  loop_control:
+    loop_var: interface_name
```

Comment:
We should also allow users to keep existing interfaces.
Otherwise `systemd_network_keep_existing_definitions` does not make a lot of sense.
We could either use the same variable or a new one.

Reply:
Done